contrib/buildrpm.sh

```diff
@@ -39,6 +42,9 @@ EOF
     shift
 done
 
+if [ $opt_no_dist -eq 1 ]; then
+    rpmmacros="$rpmmacros '--undefine=dist'"
```

Comment:
@miked-mellanox yes, it is possible with conditionals.
But it doesn't solve the issue: Mellanox packaging expect single source rpm for all OSs. If we add this check, Fedora rpms will expect a source rpm file with dist tag in the name while other will expect this file without this tag.